client/core/core.go

```diff
@@ -4481,13 +4481,47 @@ func (c *Core) Send(pw []byte, assetID uint32, value uint64, address string, sub
 		return nil, err
 	}
 
-	subject, details := c.formatDetails(TopicSendSuccess, unbip(assetID), coin)
+	sentValue := wallet.Info().UnitInfo.ConventionalString(coin.Value())
+	subject, details := c.formatDetails(TopicSendSuccess, sentValue, unbip(assetID), address, coin)
 	c.notify(newSendNote(TopicSendSuccess, subject, details, db.Success))
 
 	c.updateAssetBalance(assetID)
 	return coin, nil
 }
 
+// ValidateAddress checks that the provided address is valid.
+func (c *Core) ValidateAddress(address string, assetID uint32) bool {
+	if address == "" {
+		return false
+	}
+	wallet, found := c.wallet(assetID)
+	if !found {
+		return false
+	}
+	return wallet.Wallet.ValidateAddress(address)
+}
+
+// EstimateSendTxFee returns an estimate of the tx fee needed to send or
+// withdraw the specified amount.
+func (c *Core) EstimateSendTxFee(address string, assetID uint32, amount uint64, subtract bool) (fee uint64, isValidAddress bool, err error) {
+	if amount == 0 {
+		return 0, false, fmt.Errorf("cannot check fee for zero %s", unbip(assetID))
+	}
+	wallet, found := c.wallet(assetID)
+	if !found {
+		return 0, false, newError(missingWalletErr, "no wallet found for %s", unbip(assetID))
+	}
+	if subtract {
+		if _, isWithdrawer := wallet.Wallet.(asset.Withdrawer); !isWithdrawer {
+			return 0, false, fmt.Errorf("wallet does not support checking network fee for withdrawal")
+		}
+	}
+	if txfeeEstimator, ok := wallet.Wallet.(asset.TxFeeEstimator); ok {
+		return txfeeEstimator.EstimateSendTxFee(address, amount, c.feeSuggestionAny(assetID), subtract)
+	}
+	return 0, wallet.Wallet.ValidateAddress(address), nil
```

Comment:
How about a `WalletTrait` for `TxFeeEstimator`, and then error if called for a wallet that doesn't support it?